DataStructures/DoublyLinkedList/DoublyLinkedList.cs

```diff
@@ -212,85 +213,92 @@ public DoublyLinkedListNode<T> GetAt(int position)
             var current = Head;
             for (var i = 0; i < position; i++)
             {
-                current = current.Next;
+                current = current!.Next;
             }
 
-            return current;
+            return current ?? throw new ArgumentOutOfRangeException();
         }
 
         /// <summary>
         /// Removes the Head and replaces it with the second node in the list.
         /// </summary>
-        /// <returns>True if the node was able to be removed. False if it wasn't.</returns>
-        public bool RemoveHead()
+        public void RemoveHead()
         {
             if (Head is null)
             {
-                return false;
+                throw new InvalidOperationException();
             }
 
             Head = Head.Next;
-            Head.Previous = null;
-            Count--;
-
             if (Head is null)
             {
                 Tail = null;
                 Count = 0;
+                return;
             }
 
-            return true;
+            Head.Previous = null;
+            Count--;
         }
 
         /// <summary>
         /// Removes the last node in the list.
         /// </summary>
-        /// <returns>True if the node was able to be removed. False if it wasn't.</returns>
-        public bool Remove()
+        public void Remove()
         {
             if (Tail is null)
             {
-                return false;
+                throw new InvalidOperationException();
             }
 
             Tail = Tail.Previous;
+            if (Tail is null)
+            {
+                Head = null;
+                Count = 0;
+                return;
+            }
+
             Tail.Next = null;
             Count--;
-            return true;
         }
 
         /// <summary>
-        /// Removes an especific node.
+        /// Removes specific node.
         /// </summary>
         /// <param name="node"> Node to be removed.</param>
-        /// <returns>True if the node was able to be removed. False if it wasn't.</returns>
-        public bool RemoveNode(DoublyLinkedListNode<T> node)
+        public void RemoveNode(DoublyLinkedListNode<T> node)
         {
             if (node == Head)
             {
-                return RemoveHead();
+                RemoveHead();
+                return;
             }
 
             if (node == Tail)
             {
-                return Remove();
+                Remove();
+                return;
+            }
+
+            if (node.Previous is null || node.Next is null)
+            {
+                throw new ArgumentException();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use a constructor overloads that allows a more meaningful exception message to be provided.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4735456)